cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java

```diff
@@ -52,11 +69,45 @@ public HttpExecutorImpl(HttpClient client) {
         this.httpClient = client;
     }
 
+    public Organization getOrganization(URI uri) throws ApiGatewayException {
+        return fromSubSubunit(getSubSubUnitDtoWithMultipleEfforts(uri));
+    }
+
+    protected SearchResponse<Organization> query(URI uri) throws NotFoundException, FailedHttpRequestException {
+        HttpResponse<String> response = sendRequestMultipleTimes(addLanguage(uri)).get();
+        if (isSuccessful(response.statusCode())) {
+            try {
+                List<Organization> organizations = getOrganizations(response);
+                return new SearchResponse<Organization>(uri)
+                        .withHits(organizations)
+                        .withSize(getCount(response, organizations));
+            } catch (JsonProcessingException e) {
+                throw new FailedHttpRequestException(e.getMessage());
+            }
+        } else {
+            throw new NotFoundException(String.format(NOT_FOUND_MESSAGE_TEMPLATE, uri));
```

Comment:
Is this the only non successful case?

Reply:
From cristin we have 404 (Not found, which may be considered a success) and 400 which is error in query statement - and  it might also be any other error from the net